### PR TITLE
[AF-1392] Unable to import filesystem bare git repository in Windows

### PR DESCRIPTION
**Pull request by elguardian**
Path (String) does not work well in windows. Wrapping the string in an URI.

### CODE REVIEW

**Comment by ederign** [comment]
Also, a second question: with your change did you successfully import a project?

**Comment by elguardian** [comment]
@ederign 
Hi, That is not correct.

Not working on windows : 
Paths.get("file:/C:/Servers/PAM7/jboss-eap-7.1/bin/.niogit/system/examples-CUsersstarksDesktopDinner-Party.git/.")

working on windows:
Paths.get("C:/Servers/PAM7/jboss-eap-7.1/bin/.niogit/system/examples-CUsersstarksDesktopDinner-Party.git/.")
or 
Paths.get(new URI("file:/C:/Servers/PAM7/jboss-eap-7.1/bin/.niogit/system/examples-CUsersstarksDesktopDinner-Party.git/."))

This is the less hacky solution I could find. Keep in mind that this part of the code new URI will never fail as it is coming from a conversion URI -> String (PathUtils.getNiogitRepoPath) but because it is a checked exception try-catch needs to be in place.

Yes I tried on windows and I was able to succesfully import the project. In linux this works properly.

**Comment by elguardian** [comment]
@ederign 
Paths.get throws an exception indeed but it is not URI syntax.
please point out the changes

**Comment by porcelli** [comment]
@elguardian the correct format is `file://` - what about check if the system is running on windows and replace `file:/` by `file://`?

**Comment by elguardian** [comment]
Hi @porcelli, does not work either and I think it is not necessary as there is a proper call working for both OS.

**Comment by porcelli** [comment]
@elguardian so, please provide some unit tests

**Comment by elguardian** [comment]
@porcelli I changed importDefaultProjectTest to provide a route that will recreate the problem in windows (if executed there)

**Comment by porcelli** [comment]
@elguardian please try to create a test with a windows path, so we can avoid future regressions. use mocks if needed

**Comment by elguardian** [comment]
done @porcelli

**Comment by porcelli** [comment]
@domhanak could you please verify this dominik? thx!

**Comment by elguardian** [comment]
@porcelli any progress ?

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-backend/src/main/java/org/kie/workbench/common/screens/impl/LibraryServiceImpl.java [hunk 1]
```diff
@@ -316,12 +317,17 @@ private boolean exists(org.uberfire.java.nio.file.Path path) {
...
-                .map(matcher -> matcher.replaceFirst(""))
-                .orElse("new-project");
+        try {
+            URI repositoryURI = new URI(repositoryURL);
+            return Optional.of(repositoryURL)
+                    .map(url -> java.nio.file.Paths.get(repositoryURI))
+                    .map(path -> path.getFileName().toString())
+                    .map(fileName -> STRIP_DOT_GIT.matcher(fileName))
+                    .map(matcher -> matcher.replaceFirst(""))
+                    .orElse("new-project");
```
Comment (porcelli) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-backend/src/main/java/org/kie/workbench/common/screens/impl/LibraryServiceImpl.java]: @elguardian I just think the possible return of "new-project" don't exactly reflect what we would expect. I'd much rather prefer to use `orElseThrow`.

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-backend/src/main/java/org/kie/workbench/common/screens/impl/LibraryServiceImpl.java [hunk 2]
```diff
@@ -316,12 +317,17 @@ private boolean exists(org.uberfire.java.nio.file.Path path) {
...
+        } catch (URISyntaxException e) {
+            return "new-project";
```
Comment (porcelli) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-backend/src/main/java/org/kie/workbench/common/screens/impl/LibraryServiceImpl.java]: Same here for return the `new-project`

**Comment by porcelli** [comment]
@elguardian could you please resolve the conflicts? 
@domhanak I'd really would like to include this in the next monday community release

**Comment by elguardian** [comment]
@porcelli rebased

**Comment by domhanak** [comment]
jenkins retest

**Comment by porcelli** [comment]
@domhanak update on this?

**Comment by mareknovotny** [comment]
is this ready to merge?

**Comment by porcelli** [comment]
@mareknovotny we're waiting for QE feedback, in regards 7.11.x I already here (also waiting for QE): https://github.com/kiegroup/kie-wb-common/pull/2115

**Comment by elguardian** [comment]
this would mean a regression as the  pr downstream has been merged and not this one (master)

**Pull request merged**